### ISSUE TITLE
False-positive for concatenating strings with `"/"` in `paste()`

### ISSUE THREAD

**Issue by strengejacke** [r-lib/lintr#2674]
See following example:

```r
nominator <- "x"
denominator <- "7"
paste(nominator, denominator, sep = "/")
```

which throws following lintr-warning:

> Warning: file=R/find_transformation.R,line=68,col=15,[paste_linter] Construct file paths with file.path(...) instead of paste(..., sep = "/"). If you are using paste(sep = "/") to construct a date, consider using format() or lubridate helpers instead. Note that paste() converts empty inputs to "", whereas file.path() leaves it empty.

I see this is probably not easy to solve, but in my example, the lintr-warning was a false-positive.

**Comment by MichaelChirico** [comment]
You can also change to `paste_linter(allow_file_path = "always")`

```r
lint(text = 'paste(nominator, denominator, sep = "/")', linters = paste_linter(allow_file_path="always"))
```

Otherwise, I would just `#nolint` this case, ideally it's rare or isolated to one block/file of code.
